pkg/certsigner/signer.go

```diff
@@ -125,29 +150,57 @@ func (s *CertServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	s.mux.ServeHTTP(w, r)
 }
 
-// NewSigner returns a CertSigner object after filling in its attibutes
-// from the `Config` provided.
-func NewSigner(c Config) (*CertSigner, error) {
-	ca, err := ioutil.ReadFile(c.CACertFile)
+// newSignerCA returns a SignerCA object of PEM encoded CA cert and keys based on the profile passed.
+func newSignerCA(sc *SignerCAFiles, csr *capi.CertificateSigningRequest) (*SignerCA, error) {
+	var caCert, caKey string
+
+	profile, err := getProfile(csr)
 	if err != nil {
-		return nil, fmt.Errorf("error reading CA cert file %q: %v", c.CACertFile, err)
+		return nil, err
 	}
-	cakey, err := ioutil.ReadFile(c.CAKeyFile)
+	switch profile {
+	case "EtcdMetric":
+		if sc.MetricCAKey != "" && sc.MetricCACert != "" {
```

Comment:
why is this different from https://github.com/coreos/kubecsr/pull/24/files#diff-654ba8df3be94fe94c9404a7727d1fa2R170 ?

Reply:
We need this logic (somewhere) because we do not make the metrics flags mandatory. When EtcdMetric profile is passed it means we must make sure to sign using metrics CA. But we also need to make sure they exist. So we could make the flags mandatory but it seems metrics signer might be optional functionality. Does that answer your question?

Reply:
Other option would be to fallback on the Peer/Server CA if we do not pass a separate CA for metrics.

Reply:
Added a flag that makes this a little more straightforward.